engine/src/flutter/tools/pkg/engine_build_configs/bin/check.dart

```diff
@@ -2,34 +2,89 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'dart:io' as io show Directory, exitCode, stderr;
+import 'dart:io' as io;
 
+import 'package:args/args.dart';
 import 'package:engine_build_configs/engine_build_configs.dart';
+import 'package:engine_build_configs/src/ci_yaml.dart';
 import 'package:engine_repo_tools/engine_repo_tools.dart';
+import 'package:meta/meta.dart';
 import 'package:path/path.dart' as p;
 import 'package:platform/platform.dart';
+import 'package:source_span/source_span.dart';
+import 'package:yaml/yaml.dart' as y;
 
 // Usage:
-// $ dart bin/check.dart [/path/to/engine/src]
+// $ dart bin/check.dart
+//
+// Or, for more options:
+// $ dart bin/check.dart --help
+
+final _argParser =
+    ArgParser()
+      ..addFlag('verbose', abbr: 'v', help: 'Enable noisier diagnostic output', negatable: false)
+      ..addFlag('help', abbr: 'h', help: 'Output usage information.', negatable: false)
+      ..addOption(
+        'engine-src-path',
+        valueHelp: '/path/to/engine/src',
+        defaultsTo: Engine.tryFindWithin()?.srcDir.path,
+      );
 
 void main(List<String> args) {
-  final String? engineSrcPath;
-  if (args.isNotEmpty) {
-    engineSrcPath = args[0];
-  } else {
-    engineSrcPath = null;
-  }
+  run(
+    args,
+    stderr: io.stderr,
+    stdout: io.stdout,
+    platform: const LocalPlatform(),
+    setExitCode: (exitCode) {
+      io.exitCode = exitCode;
+    },
+  );
+}
 
-  // Find the engine repo.
-  final Engine engine;
-  try {
-    engine = Engine.findWithin(engineSrcPath);
-  } catch (e) {
-    io.stderr.writeln(e);
-    io.exitCode = 1;
+@visibleForTesting
+void run(
+  Iterable<String> args, {
+  required Platform platform,
+  required StringSink stderr,
+  required StringSink stdout,
+  required void Function(int) setExitCode,
+}) {
+  y.yamlWarningCallback = (String message, [SourceSpan? span]) {};
+
+  final argResults = _argParser.parse(args);
+  if (argResults.flag('help')) {
+    stdout.writeln(_argParser.usage);
     return;
   }
 
+  final verbose = argResults.flag('verbose');
+  void debugPrint(String output) {
+    if (!verbose) {
+      return;
+    }
+    stderr.writeln(output);
+  }
+
+  void indentedPrint(Iterable<String> errors) {
+    for (final error in errors) {
+      stderr.writeln('  $error');
+    }
+  }
+
+  final supportsEmojis = !platform.isWindows || platform.environment.containsKey('WT_SESSION');
```

Comment:
Could you check how the emojis come through in the logs? If they're garbled or placeholders, maybe you can plumb `stdout.hasTerminal` through a boolean parameter to `run()`.

Reply:
<img width="564" alt="Screenshot 2025-01-22 at 9 04 47 AM" src="https://github.com/user-attachments/assets/2609ef49-f769-470f-a9f5-70b1b782466d" />

[Looks](https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8725078500428378785/+/u/test:_Check_build_configs/stdout) good!